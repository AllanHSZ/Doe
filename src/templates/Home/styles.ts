import media from 'styled-media-query'
import styled, { css } from 'styled-components'

import { Footer } from 'components/Footer/styles'
import { Wrapper as CardButton } from 'components/CardButton/styles'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: ${theme.spacings.small};
    overflow-y: auto;
    /* padding: ${theme.spacings.xsmall}; */
    height: 100vh;

    ${Footer} {
      margin-top: min(2vh, ${theme.spacings.medium});
    }

    @media (max-width: 767px) {
      ${CardButton}:not(:first-child) {
        text-align: center;
        flex-direction: column;
      }
    }

    ${media.greaterThan('small')`
      gap: ${theme.spacings.small};
      padding: ${theme.spacings.small};

      ${Footer} {
        margin-top: min(6vh, ${theme.spacings.large});
      }
    `}

    ${media.greaterThan('medium')`
      align-items: flex-start;
      padding: ${theme.spacings.xsmall} ${theme.spacings.xlarge};
      justify-content: flex-end;

      ${Header} {
        flex: initial;
        align-items: flex-start;
      }

      ${Hello} {
        margin-top: ${theme.spacings.small};
      }

      ${Main} {
        max-width: unset;
        grid-template-areas:
          'account .'
          'donation receive'
          'connections .';
      }

    `}

    ${media.greaterThan('huge')`
      max-width: 88rem;
      margin: 0 auto;
    `}
  `}
`

export const Header = styled.header`
  ${({ theme }) => css`
    flex: 2;
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;
    font-family: ${theme.font.family_pacifico};
    color: ${theme.colors.text};
    line-height: 1.4;
  `}
`

export const Hello = styled.section`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    gap: ${theme.spacings.xxsmall};
    max-width: 30rem;
    width: 100%;
  `}
`

export const Main = styled.main`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: ${theme.spacings.xsmall};
    max-width: 30rem;

    grid-template-areas:
      'account account'
      'donation receive'
      'connections connections';

    & > *:first-child {
      grid-area: account;
    }

    & > *:nth-child(2) {
      grid-area: donation;
    }

    & > *:nth-child(3) {
      grid-area: receive;
    }
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xhuge};
  `}
`

export const Connections = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    font-size: ${theme.font.sizes.xsmall};
    font-weight: ${theme.font.light};
    grid-area: connections;
  `}
`
