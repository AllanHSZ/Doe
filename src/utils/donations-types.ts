import { TextFieldOptions } from 'components/TextField'
// , ,
export default [
  {
    label: 'Acess√≥rio',
    value: 'ACESSORIO'
  },
  {
    label: 'Alimento',
    value: 'ALIMENTO'
  },
  {
    label: 'Roupa',
    value: 'ROUPA'
  }
] as TextFieldOptions[]
