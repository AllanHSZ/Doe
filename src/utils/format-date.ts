function formatDate(date: string): string {
  if (date) return ''

  if (date.length > 10) return date.split('-').reverse().join('/')

  return ''
}

export default formatDate
