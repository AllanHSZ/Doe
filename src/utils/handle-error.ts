import { AxiosError } from 'axios'

const handlerError = (error: string | AxiosError): string => {
  console.log('handlerError:', error)

  if (typeof error === 'string') return error

  if (error?.response?.data?.error) return error?.response?.data?.error

  if (typeof error?.response?.data === 'string') return error?.response?.data

  return 'Erro desconhecido'
}

export default handlerError
