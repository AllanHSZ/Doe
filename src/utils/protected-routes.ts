import { GetServerSidePropsContext } from 'next'
import { DefaultSession, Session } from 'next-auth'
import { getSession } from 'next-auth/client'

export type MySession = {
  id: number
} & Pick<DefaultSession, 'expires' | 'user'>

async function protectedRoutes(
  context: GetServerSidePropsContext,
  session?: Session | null
) {
  if (!session) session = await getSession(context)

  if (!session?.user?.id) {
    context.res.writeHead(302, {
      Location: `/signin?callbackUrl=${context.resolvedUrl}`
    })
    context.res.end()
  }

  return session
}

export default protectedRoutes
