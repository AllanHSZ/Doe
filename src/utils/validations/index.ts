import Joi from 'joi'

const userFieldsValidations = {
  name: Joi.string().min(5).required().messages({
    'string.empty': 'Digite o seu nome.',
    'string.min': 'O nome precisa ter no mínimo 5 caracteres.'
  }),
  email: Joi.string()
    .email({ tlds: { allow: false } })
    .required()
    .messages({
      'string.empty': 'Digite o seu email.',
      'string.email': 'Email inválido.'
    }),
  password: Joi.string().min(6).required().messages({
    'string.min': 'A senha precisa ter no mínimo 6 caracteres.',
    'string.empty': 'Digite a sua senha.'
  }),
  confirm_password: Joi.string()
    .valid(Joi.ref('password'))
    .required()
    .messages({
      'any.required': 'Digite a senha novamente.',
      'any.only': 'As senhas não são iguais.'
    })
}

const addressFieldsValidations = {
  city: Joi.string().required().messages({
    'string.empty': 'Digite a cidade.'
  }),
  neighborhood: Joi.string().required().messages({
    'string.empty': 'Digite o bairro.'
  }),
  number: Joi.string().required().messages({
    'string.empty': 'Digite o número.'
  }),
  state: Joi.string().required().messages({
    'string.empty': 'Digite o estado.'
  }),
  street: Joi.string().required().messages({
    'string.empty': 'Digite a rua.'
  })
}

const donatinsFieldsValidations = {
  type: Joi.string().required().messages({
    'string.empty': 'Digite o tipo da doação.'
  }),
  description: Joi.string().required().messages({
    'string.empty': 'Digite a descrição da doação.'
  }),
  phone: Joi.string().required().messages({
    'string.empty': 'Digite um celular para contato.'
  }),
  email: Joi.string()
    .email({ tlds: { allow: false } })
    .required()
    .messages({
      'string.empty': 'Digite um email para contato.',
      'string.email': 'Email inválido.'
    })
}

export type FieldErrors = {
  [key: string]: string
}

function getFieldErrors(objError: Joi.ValidationResult) {
  const errors: FieldErrors = {}

  if (objError.error) {
    objError.error.details.forEach((err) => {
      errors[err.path.join('.')] = err.message
    })
  }

  return errors
}

type SignUpProps = {
  name?: string
  username?: string
  password?: string
  confirm_password?: string
}

export function signUpValidate(values: SignUpProps) {
  const { email: user, ...validations } = userFieldsValidations
  const schema = Joi.object({ user, ...validations })

  return getFieldErrors(schema.validate(values, { abortEarly: false }))
}

type SignInProps = {
  username?: string
  password?: string
}

export function signInValidate(values: SignInProps) {
  const { email: username, password } = userFieldsValidations
  const schema = Joi.object({ username, password })

  return getFieldErrors(schema.validate(values, { abortEarly: false }))
}

type EditUserProps = {
  name?: string
  password?: string
  confirm_password?: string
}

export function editUserValidate(values: EditUserProps) {
  const { name, password, confirm_password } = userFieldsValidations

  const schema = Joi.object({
    name,
    ...(values?.password?.trim().length ? { password, confirm_password } : {})
  })

  return getFieldErrors(
    schema.validate(
      {
        name: values.name,
        ...(values.password
          ? {
              password: values.password,
              confirm_password: values.confirm_password
            }
          : {})
      },
      { abortEarly: false }
    )
  )
}

type AddressProps = {
  city: string
  neighborhood: string
  number: string
  state: string
  street: string
}

export function addressValidate(values: AddressProps) {
  const schema = Joi.object(addressFieldsValidations)

  return getFieldErrors(schema.validate(values, { abortEarly: false }))
}

type DonationValidationProps = {
  type: string
  description: string
  phone: string
  email: string
}

export function donationValidate(values: DonationValidationProps) {
  const schema = Joi.object(donatinsFieldsValidations)

  return getFieldErrors(schema.validate(values, { abortEarly: false }))
}
