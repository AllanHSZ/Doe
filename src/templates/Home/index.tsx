import CardButton from 'components/CardButton'
import Footer from 'components/Footer'
import Logo from 'components/Logo'
import Page from 'templates/Page'

import * as S from './styles'

export type HomeProps = {
  connections: number
}

const Home = ({ connections }: HomeProps) => (
  <Page>
    <S.Wrapper>
      <S.Header>
        <S.Title>Doe</S.Title>
        <h2>Conectando pessoas</h2>
      </S.Header>

      <S.Hello>
        <Logo size="xsmall" />
        <h3>Olá, Allan</h3>
      </S.Hello>

      <S.Main>
        <CardButton
          src="/img/avatar.svg"
          alt="Avatar do usuário"
          title="Minha conta"
          size="small"
        />
        <CardButton
          src="/img/donation.svg"
          alt="Duas mãos enviando um coração vende para cima"
          message="Ajude pessoas ou causas"
          title="Doar"
        />
        <CardButton
          src="/img/receive.svg"
          alt="Duas mãos segurando um coração vende"
          message="Procure algo que você precisa"
          title="Encontrar"
        />
        <S.Connections>{connections} pessoas conectadas</S.Connections>
      </S.Main>

      <Footer />
    </S.Wrapper>
  </Page>
)

export default Home
