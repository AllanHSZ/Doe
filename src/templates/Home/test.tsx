import { render, screen } from 'utils/test-utils'

import Home from '.'

jest.mock('hooks/use-session', () => ({
  useSession: jest.fn(() => {
    return {
      session: {
        user: {
          name: 'Allan'
        }
      },
      loading: false
    }
  })
}))

jest.mock('templates/Page', () => ({
  __esModule: true,
  default: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock Page">{children}</div>
  }
}))

describe('<Home />', () => {
  it('should render correctly', () => {
    const { container } = render(<Home connections={200} />)

    expect(screen.getByRole('heading', { name: /Doe/i })).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Conectando pessoas/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Ol√°, Allan/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/200 pessoas conectadas/i)).toBeInTheDocument()

    expect(screen.getByTestId('Mock Page')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
