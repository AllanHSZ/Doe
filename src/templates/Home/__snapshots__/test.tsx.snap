// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 1.6rem;
  border: none;
  gap: 1.6rem;
  border-radius: 0.5rem;
  box-shadow: 0px 0px 8px rgba(0,0,0,0.15);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #fff;
}

.c11 {
  width: 4rem;
  height: 4rem;
}

.c14 {
  width: 10rem;
  height: 8rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.8rem;
}

.c12 h3,
.c12 h4 {
  text-align: left;
  color: #212D22;
}

.c15 {
  color: #394338;
  font-size: 1.4rem;
}

.c18 {
  color: #394338;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c19 {
  font-size: 1.2rem;
}

.c6 {
  width: 4.6rem;
  height: 3.9rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2.4rem;
  overflow-y: auto;
  height: 100vh;
}

.c0 .c17 {
  margin-top: min(2vh,3.2rem);
}

.c2 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: 'Pacifico','Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  color: #394338;
  line-height: 1.4;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.8rem;
  max-width: 30rem;
  width: 100%;
}

.c8 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 1.6rem;
  max-width: 30rem;
  grid-template-areas: 'account account' 'donation receive' 'connections connections';
}

.c8 > *:first-child {
  grid-area: account;
}

.c8 > *:nth-child(2) {
  grid-area: donation;
}

.c8 > *:nth-child(3) {
  grid-area: receive;
}

.c3 {
  font-size: 8.0rem;
}

.c16 {
  color: #394338;
  font-size: 1.2rem;
  font-weight: 300;
  grid-area: connections;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.4rem;
  padding: 0.8rem;
  margin-left: auto;
  cursor: pointer;
}

@media (max-width:767px) {
  .c0 .c9:not(:first-child) {
    text-align: center;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:450px) {
  .c0 {
    gap: 2.4rem;
    padding: 2.4rem;
  }

  .c0 .c17 {
    margin-top: min(6vh,4.0rem);
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    gap: 4.0rem;
    padding: 1.6rem 4.8rem;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }

  .c0 .c1 {
    -webkit-flex: initial;
    -ms-flex: initial;
    flex: initial;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }

  .c0 .c4 {
    margin-top: 2.4rem;
  }

  .c0 .c7 {
    max-width: unset;
    grid-template-areas: 'account .' 'donation receive' 'connections .';
  }
}

@media (min-width:1440px) {
  .c0 {
    max-width: 88rem;
    margin: 0 auto;
  }
}

<div
  data-testid="Mock Page"
>
  <div
    class="c0"
  >
    <header
      class="c1 c2"
    >
      <h1
        class="c3"
      >
        Doe
      </h1>
      <h2>
        Conectando pessoas
      </h2>
    </header>
    <section
      class="c4 c5"
    >
      <div
        class="c6"
      >
        <svg
          aria-label="Doe"
          fill="none"
          height="100%"
          role="img"
          viewBox="0 0 107 107"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            filter="url(#logo1)"
          >
            <g
              filter="url(#logo2)"
            >
              <path
                d="M109.972 35.685c-1.875-27.12-31.403-37.407-50.619-21.509L70.602 24.93c9.373 13.56-2.813 24.314-16.873 16.833L30.763 65.142c-4.218 5.143.469 12.624 7.499 8.884l26.247-25.25 2.812 2.339-25.778 26.651c-3.281 5.611 2.343 11.222 8.905 7.482l25.31-25.25 3.28 2.338L53.26 88.521c-4.218 5.61 2.812 12.624 8.437 7.948l40.307-39.743c8.905-11.69 7.968-21.041 7.968-21.041Z"
                fill="#04D361"
              />
              <path
                d="m50.917 38.958-7.968-8.884-23.435 23.378c-5.624 5.144 1.875 15.43 8.905 8.417l22.498-22.911Z"
                fill="#04D361"
              />
              <path
                d="M54.666 13.709C49.511 9.5 34.981-.786 16.234 12.773-3.576 27.1 7.327 52.985 14.826 58.596c-.937-2.806 1.875-7.014 3.281-8.417l22.497-22.91-7.499-7.014 2.344-1.87 19.685 19.637C65.446 48.31 75.757 34.75 67.32 26.334c0 0-7.345-8.29-12.655-12.624Z"
                fill="#04D361"
              />
              <path
                d="M109.972 35.685c-1.875-27.12-31.403-37.407-50.619-21.509L70.602 24.93c9.373 13.56-2.813 24.314-16.873 16.833L30.763 65.142c-4.218 5.143.469 12.624 7.499 8.884l26.247-25.25 2.812 2.339-25.778 26.651c-3.281 5.611 2.343 11.222 8.905 7.482l25.31-25.25 3.28 2.338L53.26 88.521c-4.218 5.61 2.812 12.624 8.437 7.948l40.307-39.743c8.905-11.69 7.968-21.041 7.968-21.041Z"
                stroke="#000"
                stroke-width=".5"
              />
              <path
                d="m50.917 38.958-7.968-8.884-23.435 23.378c-5.624 5.144 1.875 15.43 8.905 8.417l22.498-22.911Z"
                stroke="#000"
                stroke-width=".5"
              />
              <path
                d="M54.666 13.709C49.511 9.5 34.981-.786 16.234 12.773-3.576 27.1 7.327 52.985 14.826 58.596c-.937-2.806 1.875-7.014 3.281-8.417l22.497-22.91-7.499-7.014 2.344-1.87 19.685 19.637C65.446 48.31 75.757 34.75 67.32 26.334c0 0-7.345-8.29-12.655-12.624Z"
                stroke="#000"
                stroke-width=".5"
              />
            </g>
          </g>
          <defs>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="100.5"
              id="logo1"
              width="254.17"
              x=".75"
              y="5.75"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecolormatrix
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
              />
              <feblend
                in2="BackgroundImageFix"
                result="effect1_dropShadow"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_dropShadow"
                result="shape"
              />
            </filter>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="96.5"
              id="logo2"
              width="105.5"
              x="4.75"
              y="5.75"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <feblend
                in="SourceGraphic"
                in2="BackgroundImageFix"
                result="shape"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecomposite
                in2="hardAlpha"
                k2="-1"
                k3="1"
                operator="arithmetic"
              />
              <fecolormatrix
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
              />
              <feblend
                in2="shape"
                result="effect1_innerShadow"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset />
              <fegaussianblur
                stdDeviation="5"
              />
              <fecomposite
                in2="hardAlpha"
                k2="-1"
                k3="1"
                operator="arithmetic"
              />
              <fecolormatrix
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
              />
              <feblend
                in2="effect1_innerShadow"
                result="effect2_innerShadow"
              />
            </filter>
          </defs>
        </svg>
      </div>
      <h3>
        Ol√°, 
        Allan
      </h3>
    </section>
    <main
      class="c7 c8"
    >
      <a
        class="c9 c10"
      >
        <img
          alt="Avatar do usu√°rio"
          class="c11"
          src="/img/avatar.svg"
        />
        <div
          class="c12"
        >
          <h4>
            Minha conta
          </h4>
        </div>
        <div
          class="c13"
          role="button"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
            />
            <polyline
              points="16 17 21 12 16 7"
            />
            <line
              x1="21"
              x2="9"
              y1="12"
              y2="12"
            />
          </svg>
        </div>
      </a>
      <a
        class="c9 c10"
      >
        <img
          alt="Duas m√£os enviando um cora√ß√£o vende para cima"
          class="c14"
          src="/img/donation.svg"
        />
        <div
          class="c12"
        >
          <h3>
            Doar
          </h3>
          <p
            class="c15"
          >
            Ajude pessoas ou causas
          </p>
        </div>
      </a>
      <a
        class="c9 c10"
      >
        <img
          alt="Duas m√£os segurando um cora√ß√£o vende"
          class="c14"
          src="/img/receive.svg"
        />
        <div
          class="c12"
        >
          <h3>
            Encontrar
          </h3>
          <p
            class="c15"
          >
            Procure algo que voc√™ precisa
          </p>
        </div>
      </a>
      <p
        class="c16"
      >
        200
         pessoas conectadas
      </p>
    </main>
    <footer
      class="c17 c18"
    >
      <p
        class="c19"
      >
        Doe ¬© 2021, Nenhum direito reservado.
      </p>
    </footer>
  </div>
</div>
`
